backend/src/schema/resolvers/posts.spec.js

```diff
@@ -357,6 +374,78 @@ describe('CreatePost', () => {
         })
       })
     })
+
+    describe('pinned posts', () => {
+      beforeEach(async () => {
+        variables = { ...variables, categoryIds: ['cat9', 'cat27', 'cat15'], pinned: true }
+      })
+      describe('users cannot create pinned posts', () => {
+        it('throws authorization error', async () => {
+          await expect(mutate({ mutation: createPostMutation, variables })).resolves.toMatchObject({
+            errors: [{ message: 'Not Authorised!' }],
+            data: { CreatePost: null },
+          })
+        })
+      })
+
+      describe('moderator cannot create pinned posts', () => {
+        let moderator
+        beforeEach(async () => {
+          moderator = await user.update({ role: 'moderator', updatedAt: new Date().toISOString() })
+          authenticatedUser = await moderator.toJson()
+        })
+
+        it('throws authorization error', async () => {
+          await expect(mutate({ mutation: createPostMutation, variables })).resolves.toMatchObject({
+            errors: [{ message: 'Not Authorised!' }],
+            data: { CreatePost: null },
+          })
+        })
+      })
+
+      describe('admin can create pinned posts', () => {
+        let admin
+        beforeEach(async () => {
+          admin = await user.update({
+            role: 'admin',
+            name: 'Admin',
+            updatedAt: new Date().toISOString(),
+          })
+          authenticatedUser = await admin.toJson()
+          variables = {
+            ...variables,
+            title: 'pinned post',
+            content:
+              'this is super important for the community and we promise not to have too many',
+          }
+        })
+
+        it('throws authorization error', async () => {
+          const expected = {
+            data: {
+              CreatePost: {
+                title: 'pinned post',
+                content:
+                  'this is super important for the community and we promise not to have too many',
+                author: {
+                  name: 'Admin',
+                },
+                pinnedBy: {
+                  id: 'current-user',
+                  name: 'Admin',
+                  role: 'admin',
+                },
+              },
+            },
+            errors: undefined,
+          }
+
+          await expect(mutate({ mutation: createPostMutation, variables })).resolves.toMatchObject(
+            expected,
+          )
```

Comment:
I would add a test case here to ensure we have only one pinned post at a time. E.g. when you pin a post and there was another pinned post already, the previously pinned post gets unpinnned.